Library/Homebrew/os/linux/glibc.rb

```diff
@@ -20,6 +20,16 @@ def system_version
         @system_version = Version.new version
       end
 
+      def version
+        return @version if @version
+
+        ldd = HOMEBREW_PREFIX/"opt/glibc/bin/ldd"
+        version = Utils.popen_read(ldd, "--version")[/ (\d+\.\d+)/, 1] if ldd.executable?
+        return system_version unless version
+
+        @version = Version.new version
```

Comment:
```suggestion
        @version ||= begin
          ldd = HOMEBREW_PREFIX/"opt/glibc/bin/ldd"
          version = Utils.popen_read(ldd, "--version")[/ (\d+\.\d+)/, 1] if ldd.executable?
          if version
            Version.new version
          else
            system_version
          end
```

Otherwise it's not going to cache the `ldd` exec/filesystem access and instead shell out every time it returns `system_version` (which is undesirable). I'd suggest the same thing be done for `system_version` above too for the same reasons.

Reply:
https://github.com/Homebrew/brew/blob/master/Library/Homebrew/os/linux/glibc.rb#L15
`system_version` is cached. Can you elaborate?

Reply:
`@version` is never set if brewed glibc isn't installed, so it keep checking for brewed `glibc` on every invokation.

Reply:
Ah, I see what you mean now, when the brewed `ldd` is not present. Thanks, Mike.